### DISCUSSION TITLE
Possibility to use variables for Maven dependencies in m2e PDE integration

### DISCUSSION THREAD

**Discussion by vogella** [eclipse-m2e/m2e-core#584] [General]
It is possible to define and use variables in the target platform for the Maven dependencies?
For example, I have a implementation in which I have multiple libraries with the same version.

```
<dependency>
    <groupId>mygroup</groupId>
    <artifactId>myartifact1</artifactId>
    <version>5.8.2</version>
</dependency>
<dependency>
    <groupId>mygroup</groupId>
    <artifactId>myartifact2</artifactId>
    <version>5.8.2</version>
</dependency>
```

**Comment by laeubi** [comment]
This is not yet supported see [Bug 578431](https://bugs.eclipse.org/bugs/show_bug.cgi?id=578431)

**Reply by HannesWell** [reply]
As I have just written in the mentioned bug, the feature suggested there and the one of this discussion are both features I want to support to be implement. :)

One question I have is if it is sufficient to have only the target properties that are visible to all locations or if there is a need for maven-location specific properties (that would be in Maven-properties style)?
I think the latter is not necessary and would make thinks just more complicated.

**Reply by laeubi** [reply]
The target specific properties, in the context of tycho, could be overridden by the maven-user properties but as targets are not connected to any project context it might be only the global properties are used (similar to the reason why repositories are not possible per se).

Given that I think we should only support one set of variables and those must be override by `-D<name>=<value>`
